### ISSUE TITLE
Compiler error: NoSuchMethod Kotlin 1.7.0-RC

### ISSUE THREAD

**Issue by MariusVolkhart** [drewhamilton/Poko#75]
I get the following error when running my gradle build after updating to:

    Kotlin: 1.7.0-RC
    KSP: 1.7.0-RC-1.0.5
    Poko: 0.10.0

(I know it's pre-release, but figured I'd raise awareness that there appears to be an API change coming)

```java
e: java.lang.NoSuchMethodError: 'boolean org.jetbrains.kotlin.ir.declarations.IrClass.isInline()'
	at dev.drewhamilton.poko.ir.PokoMembersTransformer.isPokoClass(PokoMembersTransformer.kt:104)
	at dev.drewhamilton.poko.ir.PokoMembersTransformer.visitFunctionNew(PokoMembersTransformer.kt:77)
	at org.jetbrains.kotlin.backend.common.IrElementTransformerVoidWithContext.visitFunction(IrElementTransformerVoidWithContext.kt:83)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitConstructor(IrElementTransformerVoid.kt:75)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitConstructor(IrElementTransformerVoid.kt:76)
	at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitConstructor(IrElementTransformerVoid.kt:24)
	at org.jetbrains.kotlin.ir.declarations.IrConstructor.accept(IrConstructor.kt:21)
```

The code in question is essentially,

```kotlin
public sealed interface Animal {
    public val name: String 
}

@Poko
public class Dog(override val name: String) : Animal
```

**Comment by drewhamilton** [comment]
Thanks for reporting. I put up #77, but tests are failing because kotlin-compile-testing needs to be updated first. See tschuchortdev/kotlin-compile-testing#268.

**Comment by drewhamilton** [comment]
Merged #77 to main; give the snapshot a try
